xlators/cluster/dht/src/dht-common.c

```diff
@@ -11195,22 +11196,28 @@ dht_notify(xlator_t *this, int event, void *data, ...)
 }
 
 int
-dht_inode_ctx_layout_get(inode_t *inode, xlator_t *this, dht_layout_t **layout)
+dht_inode_ctx_layout_get(inode_t *inode, xlator_t *this, dht_layout_t **layout,
+                         int ref)
```

Comment:
This is an invitation to a race issue. We cannot return a layout without also adding a reference.

Reply:
Currently we are not taking reference in every case while we do call dht_inode_ctx_layout_get, Do you want we need to call reference in every case and need to unref in parent_function after get the layout.

Reply:
If we return from `dht_inode_ctx_layout_get()` with a valid layout but without a reference, nothing prevents that this layout may be destroyed before it's used by the caller, so yes, we need to correctly track reference counting when the structure is used. Only valid case is when `layout` is `NULL` (meaning that the layout won't be used. The caller only wants to test if it exist). However I'm not sure if this case is very useful because as soon as the function returns, the presence/absence of a layout can already be stale, but that's probably another discussion.

Reply:
The dht takes reference only where it uses the layout to validate existing layout on backend, there are some places where it fetch the layout to check the value but still it is good we can take reference and unref in parent function. I will update in next patch.

Reply:
done